core/state_transition.go

```diff
@@ -245,7 +265,9 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	st.gas -= gas
 
 	// Check clause 6
-	if msg.Value().Sign() > 0 && !st.evm.CanTransfer(st.state, msg.From(), msg.Value()) {
+	if msg.IsAA() && msg.Value().Sign() != 0 {
```

Comment:
Do we need the `msg.Value().Sign() != 0`? Technically the message does not have value if it is AA and so this field is irrelevant?

Reply:
I just want to avoid errors further into execution where it would attempt to transfer any nonzero amount from the `ENTRY_POINT` to the AA contract and so on...